stan/math/rev/core/operator_divide_equal.hpp

```diff
@@ -7,13 +7,18 @@
 
 namespace stan {
 namespace math {
-inline var& var::operator/=(var b) {
+template <typename T>
+template <typename S, require_convertible_t<S&, promote_args_t<T>>*>
```

Comment:
This `require_convertible_t` logic... Is this so a `var_type<float>` can promote to a `var_type<double>`?

So the matrix equivalents would be between `MatrixXf` and `MatrixXd`?

Reply:
Yeah so my intent here was that we could do `var_value<double> + var_value<float>`, but Eigen doesn't even allow that so maybe the best thing here is to get rid of the template and only have it work with var_values of the same type.

Reply:
> only have it work with var_values of the same type.

I'm happy to go with that.

`typename stan::require_vt_floating_point<T>::type` and `stan::require_floating_point_t<T>` should both be the same thing for doubles, right?

Reply:
Oh I just drop in `require_vt_floating_point<T>` for `require_floating_point_t<T>` no extra stuff necessary.